File: models/user_model.py
from typing import Optional
from sqlmodel import Field, SQLModel

class UserModel(SQLModel, table=True):
    __tablename__: str = "users"

    id: Optional[int] = Field(default=None, primary_key=True)
    userName: Optional[str]
    email: Optional[str]
    name: Optional[str]    
    lastName: Optional[str]
    profileImageUrl: Optional[str]
    bio: Optional[str]    
    gender: Optional[str] = "Not Specified"

class UserGithubModel(SQLModel, table=False):
    id: Optional[int]
    userName: Optional[str]
    email: Optional[str]
    name: Optional[str]  
    lastName: Optional[str]
    profileImageUrl: Optional[str]
    bio: Optional[str]
    gender: Optional[str]
    followers: Optional[int]
    following: Optional[int]
    public_repos: Optional[int]
    url_public: Optional[str]

File: src/models/user.model.ts
import { Gender, Prisma } from "@prisma/client";

export interface User {
  id: string;
  username: string;
  name: string;
  lastName?: string | null;
  profileImageUrl?: string | null;
  bio?: string | null;
  email: string;
  gender?: Gender;
}

export type CreateUserDto = Prisma.UserCreateInput;
export type UpdateUserDto = Prisma.UserUpdateInput;
